handler/enqueue_annotation_test.go

```diff
@@ -0,0 +1,432 @@
+					Name:      "faz",
+					Annotations: map[string]string{
+						NamespacedNameAnnotation: "AppService",
+						TypeAnnotation:           schema.GroupKind{Group: "Pods", Kind: "core"}.String(),
+					},
+				},
+			}
+
+			evt := event.CreateEvent{
+				Object: repl,
+				Meta:   repl.GetObjectMeta(),
+			}
+
+			instance.Create(evt, q)
+			Expect(q.Len()).To(Equal(1))
+
+			i, _ := q.Get()
+			Expect(i).To(Equal(reconcile.Request{
+				NamespacedName: types.NamespacedName{Namespace: "", Name: "AppService"}}))
+		})
+
+		It("should enqueue a Request for a object that is cluster scoped which has the annotations", func() {
+			var nd *corev1.Node
+
+			nd = &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node-1",
+					Annotations: map[string]string{
+						NamespacedNameAnnotation: "myapp",
+						TypeAnnotation:           schema.GroupKind{Group: "ReplicaSet", Kind: "apps"}.String(),
+					},
+				},
+			}
+
+			instance = EnqueueRequestForAnnotation{Type: schema.GroupKind{Group: "ReplicaSet", Kind: "apps"}}
+
+			evt := event.CreateEvent{
+				Object: nd,
+				Meta:   nd.GetObjectMeta(),
+			}
+
+			instance.Create(evt, q)
+			Expect(q.Len()).To(Equal(1))
+
+			i, _ := q.Get()
+			Expect(i).To(Equal(reconcile.Request{
+				NamespacedName: types.NamespacedName{Namespace: "", Name: "myapp"}}))
+		})
+
+		It("should not enqueue a Request for a object that is cluster scoped which does not have annotations", func() {
+			var nd *corev1.Node
+
+			nd = &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{Name: "node-1"},
+			}
+
+			instance = EnqueueRequestForAnnotation{Type: nd.GetObjectKind().GroupVersionKind().GroupKind()}
+			evt := event.CreateEvent{
+				Object: nd,
+				Meta:   nd.GetObjectMeta(),
+			}
+
+			instance.Create(evt, q)
+			Expect(q.Len()).To(Equal(0))
+		})
+	})
+
+	Describe("EnqueueRequestForAnnotation.SetWatchOwnerAnnotation", func() {
+		It("should add the watch owner annotations without losing existing ones", func() {
+			var nd *corev1.Node
+			nd = &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node-1",
+					Annotations: map[string]string{
+						"my-test-annotation": "should-keep",
+					},
+				},
+			}
+
+			err := SetOwnerAnnotation(podOwner, nd, schema.GroupKind{Group: "Pods", Kind: "core"})
+			Expect(err).To(BeNil())
+
+			expected := map[string]string{
+				"my-test-annotation":     "should-keep",
+				NamespacedNameAnnotation: fmt.Sprintf("%v/%v", podOwner.GetNamespace(), podOwner.GetName()),
+				TypeAnnotation:           schema.GroupKind{Group: "Pods", Kind: "core"}.String(),
+			}
+
+			Expect(len(nd.GetAnnotations())).To(Equal(3))
+			Expect(nd.GetAnnotations()).To(Equal(expected))
+		})
+
+		It("should return error when the owner Group or Kind is not informed", func() {
+			var nd *corev1.Node
+			nd = &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node-1",
+				},
+			}
+
+			err := SetOwnerAnnotation(podOwner, nd, schema.GroupKind{Group: "", Kind: "core"})
```

Comment:
See that it is a helper. so, we need to update the places in SDK that is using the annotation to use the helper instead which will be cleaner.